lib/common/firebase/firebase_module.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:firebase_core/firebase_core.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/material.dart';
+import 'package:get_it/get_it.dart';
+
+final getIt = GetIt.instance;
+
+Future<void> setupFirebase() async {
+  WidgetsFlutterBinding.ensureInitialized();
+
+  await Firebase.initializeApp(
+    options: const FirebaseOptions(
+      apiKey: "AIzaSyAzv3b5Txc0jPkTTBXXq9FqUE5YNXt7p8c",
```

Comment:
Copy the firebase_module structure from discord, where I use injectable @module and @singleton to automatically register the firebase components into the app, for them to be automatically passed to the constructors when using injectable